container.go

```diff
@@ -436,17 +436,11 @@ func (c *Container) fetchState(cmd string) (State, error) {
 	if cmd == "" {
 		return State{}, fmt.Errorf("Cmd cannot be empty")
 	}
-
-	state, err := c.pod.storage.fetchPodState(c.pod.id)
-	if err != nil {
-		return State{}, err
-	}
-
-	if state.State != StateRunning {
+	if c.pod.state.State != StateRunning {
```

Comment:
Could you explain why we're no longer fetching the state from storage ?

Reply:
before this line being executed I changed, the pod state by using `p.startSetStates()` which changes the state only in memory but in in storage, I think we should relay in the current state of the pod object that is in memory. Otherwise I could do `p.startSetStates()`  + a method to save the pod state in storage ( Taking advantage of this comment, what is that method ? )

Reply:
> before this line being executed I changed, the pod state by using `p.startSetStates()`

Where do you do that ?

> `p.startSetStates()` + a method to save the pod state in storage ( Taking advantage of this comment, what is that method ? )

`p.startSetStates() ` already stores the pod state in storage.

Reply:
>Where do you do that ?

https://github.com/containers/virtcontainers/pull/360/files#diff-c7bc83b548b0dcb1a7e055bb0d07779cR793
```Go
	// Pod is started
	if err := p.startSetStates(); err != nil {
		return err
	}

	for _, c := range p.containers {
		if err := c.start(); err != nil {
			return err
		}
	}
```
1. Start pod.
2. Change the state of pod to started calling `p.startSetStates`
3. For each container in pod : do `cotainer.start()` 
`containter.start()` will call `container.fetchState()` which require the pod state being in `running` state. But I got an error saying pod was not running. 

Let me check again and see if it is working correctly by getting from `storage`.